test/e2e/utils/utils.go

```diff
@@ -80,8 +80,9 @@ func getBoskosProject(resourceType string) *common.Resource {
 			}
 			if p == nil {
 				glog.Warningf("boskos does not have a free %s at the moment", resourceType)
+			} else {
+				return p
 			}
-			return p
```

Comment:
not convinced this fixes it as `if err != nil || p== nil` we will still return `p`. Maybe there should be a `continue` in each of the error blocks so it will go to the next `tick/timeout` loop?